core/src/test/java/org/apache/calcite/sql2rel/RelFieldTrimmerTest.java

```diff
@@ -544,4 +548,36 @@ public static Frameworks.ConfigBuilder config() {
         + "        LogicalTableScan(table=[[scott, EMP]])\n";
     assertThat(trimmed, hasTree(expected));
   }
+
+  @Test void testLogicalCorrelateFieldTrimmer() {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final Holder<@Nullable RexCorrelVariable> v = Holder.empty();
+    RelNode root = builder.scan("EMP")
+        .projectPlus(builder.call(SqlStdOperatorTable.PLUS, builder.field(0), builder.field(0)))
+        .variable(v::set)
+        .scan("DEPT")
+        .filter(
+            builder.equals(builder.field(0),
+                builder.call(
+                    SqlStdOperatorTable.PLUS,
+                    builder.literal(10),
+                    builder.field(v.get(), "DEPTNO"))))
+        .correlate(JoinRelType.LEFT, v.get().id, builder.field(2, 0, "DEPTNO"))
+        .aggregate(builder.groupKey("ENAME"), builder.max(builder.field("EMPNO")))
+        .build();
+
+    final RelFieldTrimmer fieldTrimmer = new RelFieldTrimmer(null, builder);
+    final RelNode trimmed = fieldTrimmer.trim(root);
+    final String expected = ""
```

Comment:
a bit difficult to see what the trimmer has done.
perhaps if you have the expected tree before the trim it will be easier?
then you can also assert that they are different

Reply:
I've updated the testcase to use an `Uncollect` instead - and also added the before-trim plan to the test.

Reply:
Can you add a more generalized example, where both left and right are SCAN and are correlate by some field, and both left and right be trimmed.

Reply:
I've added another test which removes an unecessarily projected expression from both sides